### PR TITLE
Add maxcount parameter to SteamDirectory.LoadAsync

### LINKED ISSUE #560: Allow to specify maxcount in SteamDirectory.LoadAsync

### PR DESCRIPTION
**Pull request by yaakov-h**
Closes #560.

cc @xpaw

### CODE REVIEW

**Comment by xPaw** [comment]
No `(SteamConfiguration configuration, int maxNumServers)` overload?

**Comment by yaakov-h** [comment]
I did consider it, but with C# 7.1 you can easily call `.LoadAsync(configuration, maxNumServers, default)`. I don't feel there's much point adding an additional overload that just passes through a `default(CancellationToken)`/`CancellationToken.None`.

File: SteamKit2/SteamKit2/Steam/WebAPI/SteamDirectory.cs [hunk 1]
```diff
@@ -42,42 +53,44 @@ public static Task<IReadOnlyCollection<ServerRecord>> LoadAsync( SteamConfigurat
...
+            if ( maxNumServers.HasValue )
+            {
+                args[ "maxcount" ] = maxNumServers.Value.ToString( CultureInfo.InvariantCulture );
+            }
+
             cancellationToken.ThrowIfCancellationRequested();
 
-            var task = directory.CallAsync( HttpMethod.Get, "GetCMList", version: 1, args: args );
-            return task.ContinueWith( t =>
+            var response = await directory.CallAsync( HttpMethod.Get, "GetCMList", version: 1, args: args ).ConfigureAwait( false );
```
Comment (VoiDeD) [inline SteamKit2/SteamKit2/Steam/WebAPI/SteamDirectory.cs]: Shame we can't pass the cancellation token further. Maybe worth expanding WebAPI's `AsyncInterface` someday.

**Pull request merged**